ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/networking/libp2p/rpc/BeaconBlocksByRootRequestMessage.java

```
@@ -17,55 +17,66 @@
 
 import com.google.common.base.MoreObjects;
 import java.util.List;
-import java.util.Objects;
 import org.apache.tuweni.bytes.Bytes32;
 import tech.pegasys.teku.ssz.SSZTypes.SSZContainer;
-import tech.pegasys.teku.ssz.SSZTypes.SSZList;
-import tech.pegasys.teku.ssz.SSZTypes.SSZMutableList;
+import tech.pegasys.teku.ssz.backing.ListViewRead;
+import tech.pegasys.teku.ssz.backing.containers.Container1;
+import tech.pegasys.teku.ssz.backing.containers.ContainerType1;
+import tech.pegasys.teku.ssz.backing.tree.TreeNode;
+import tech.pegasys.teku.ssz.backing.type.BasicViewTypes;
+import tech.pegasys.teku.ssz.backing.type.ListViewType;
+import tech.pegasys.teku.ssz.backing.view.BasicViews.Bytes32View;
+import tech.pegasys.teku.ssz.backing.view.ViewUtils;
+import tech.pegasys.teku.ssz.sos.SszTypeDescriptor;
 
-public class BeaconBlocksByRootRequestMessage implements RpcRequest, SSZContainer {
+public class BeaconBlocksByRootRequestMessage
+    extends Container1<BeaconBlocksByRootRequestMessage, ListViewRead<Bytes32View>>
+    implements RpcRequest, SSZContainer {
 
-  private final SSZMutableList<Bytes32> blockRoots =
-      SSZList.createMutable(Bytes32.class, MAX_REQUEST_BLOCKS);
+  public static class BeaconBlocksByRootRequestMessageType
+      extends ContainerType1<BeaconBlocksByRootRequestMessage, ListViewRead<Bytes32View>> {
 
-  public BeaconBlocksByRootRequestMessage(final List<Bytes32> blockRoots) {
-    this.blockRoots.addAll(blockRoots);
-  }
+    public BeaconBlocksByRootRequestMessageType() {
+      super(
+          "BeaconBlocksByRootRequestMessage",
+          namedType(
+              "blockRoots", new ListViewType<>(BasicViewTypes.BYTES32_TYPE, MAX_REQUEST_BLOCKS)));
+    }
```

Note that block by root requests aren't actually encoded as an SSZ list - they just concatenate the block roots. We have `BeaconBlocksByRootRequestMessageEncoder` to handle this and never actually SSZ encode or decode them.  They only implemented `SSZContainer` to make the type system happy originally (and it looks like that's not actually required anymore).

---

Yep, I also missed that case during refactoring, but it was caught here: [LengthPrefixedEncodingTest.encodePayload_shouldEncodeBlocksByRootRequest()](https://app.circleci.com/pipelines/github/ConsenSys/teku/10143/workflows/8cf9c9fe-ae48-4485-a130-dbffeb6e9334/jobs/63995/tests#failed-test-6) . 

Fixed with this commit: 076c75cd

---

>  They only implemented SSZContainer to make the type system happy originally (and it looks like that's not actually required anymore).

Yeah, right. We may now do this in a bit more canonical manner: 20e6eb6

---

That reminds me to check that on deserialization we are failing fast without reading the whole list when ssz is too long for that list. 
Fixed some cases and added a test here: 3a298b8